File: python/generate_api_key.py
import os
import base64
from cryptography.fernet import Fernet
import secrets

# Load the encryption key from environment variables or your config
ENCRYPTION_KEY = os.getenv('ENCRYPTION_KEY')

if not ENCRYPTION_KEY:
    raise ValueError("No ENCRYPTION_KEY set for environment")

fernet = Fernet(ENCRYPTION_KEY.encode())

def encrypt(data: str) -> str:
    return fernet.encrypt(data.encode()).decode()

def generate_api_key() -> str:
    # Generate a random API key and ensure it is base64 encoded with padding
    api_key = secrets.token_bytes(32)
    return base64.b64encode(api_key).decode('utf-8')

def main():
    api_key = generate_api_key()
    encrypted_api_key = encrypt(api_key)
    
    print(f"Generated API Key: {api_key}")
    print(f"Encrypted API Key: {encrypted_api_key}")

if __name__ == "__main__":
    main()

File: node/generate_api_key.js
const crypto = require("crypto");

// Load the encryption key from environment variables
const ENCRYPTION_KEY = Buffer.from(process.env.ENCRYPTION_KEY, "base64");

if (!ENCRYPTION_KEY) {
  throw new Error("No ENCRYPTION_KEY set for environment");
}

function encrypt(text) {
  const iv = crypto.randomBytes(16);
  const cipher = crypto.createCipheriv("aes-256-cbc", ENCRYPTION_KEY, iv);
  let encrypted = cipher.update(text, "utf8", "base64");
  encrypted += cipher.final("base64");

  return {
    iv: iv.toString("base64"),
    content: encrypted,
  };
}

function generateApiKey() {
  // Generate a random API key
  return crypto.randomBytes(32).toString("base64"); // Correct encoding method
}

function main() {
  const apiKey = generateApiKey();
  const encryptedApiKey = encrypt(apiKey);

  console.log("Generated API Key:", apiKey);
  console.log("Encrypted API Key:", JSON.stringify(encryptedApiKey));
}

main();
